README.md

```diff
@@ -1,11 +1,27 @@
 # ismp-rs ![Unit Tests](https://github.com/polytope-labs/ismp-rs/actions/workflows/ci.yml/badge.svg)
 
-Rust implementation of the Interoperable State Machine Protocol. This project is This project is [funded by the web3 foundation](https://github.com/w3f/Grants-Program/blob/master/applications/ismp.md).
+Rust implementation of the Interoperable State Machine Protocol. This project is [funded by the web3 foundation](https://github.com/w3f/Grants-Program/blob/master/applications/ismp.md).
 
 ## Overview
 
 This repo provides an implementation of the neccessary components laid out in the [ISMP spec](https://github.com/polytope-labs/ismp).
 
+## Testing and Testing Guide
+Requires [Rust](https://www.rust-lang.org/tools/install), along with it's [nightly](https://rust-lang.github.io/rustup/concepts/channels.html#:~:text=it%20just%20run-,rustup%20toolchain%20install%20nightly,-%3A) version.
+
+Run ismp-rs test suite;
+
+```bash
+cd ismp-testsuite/
+cargo test
+```
+
+## Run Test in Docker
+```bash
+docker run --memory="24g" --rm --user root -v "$PWD":/app -w /app rust:latest cargo test --release --manifest-path=./ismp-testsuite/Cargo.to
```

Comment:
Do we need up to `24gb` to run the test, I think `8g` should be sufficient, it's just a crate